sklearn/pipeline.py

```diff
@@ -537,8 +528,7 @@ def make_pipeline(*steps, **kwargs):
     ----------
     *steps : list of estimators,
 
-    memory : Instance of sklearn.externals.joblib.Memory or string, optional \
-            (default=None)
+    memory : joblib.Memory-like, optional (default=None)
```

Comment:
I think we need to say string is an option in the type spec or we make this seem harder to use than it is. If we're short of space, I don't feel default=None adds anything.

Reply:
@lesteve  suggested this to me, WDYT should be here?

Reply:
Agreed, did not spot it.

Reply:
Should I add `or string` ?

Reply:
drop (default=None) for sure. I would add or string since it would work if you pass one.

Reply:
What's the reason for dropping `default=None` ? We are not short of space.

Reply:
It's something we're not very consistent with here. I find that `default=None` is very uninformative:
1. Default values (but not their meaning) are mentioned in the function signature, and key projects like numpy etc do not mention them here in the type spec.
2. "optional" and "default=..." are redundant together.
3. "default=None" has a different semantics (i.e. it does something different) everywhere you see it: sometimes it means "determine it automatically"; sometimes "switch the feature off"; sometimes "all"; sometimes "none".

So I much prefer not to say "default=None", where "optional" already implies (and can be made explicit in the description) that the user can live without the feature.

But you certainly should state that passing a string is one way to specify `memory`.